modules/services/hypridle.nix

```
@@ -0,0 +1,82 @@
+{ config, lib, pkgs, ... }:
+with lib;
+let
+
+  cfg = config.services.hypridle;
+in {
+  meta.maintainers = [ maintainers.khaneliman maintainers.fufexan ];
+
+  options.services.hypridle = {
+    enable = mkEnableOption "Hypridle, Hyprland's idle daemon";
+
+    package = mkOption {
+      description = "The hypridle package";
+      type = types.package;
+      default = pkgs.hypridle;
+    };
+
+    settings = lib.mkOption {
+      type = with lib.types;
+        let
+          valueType = nullOr (oneOf [
+            bool
+            int
+            float
+            str
+            path
+            (attrsOf valueType)
+            (listOf valueType)
+          ]) // {
+            description = "Hypridle configuration value";
+          };
+        in valueType;
+      default = { };
+      description = ''
+        Hypridle configuration written in Nix. Entries with the same key
+        should be written as lists. Variables' and colors' names should be
+        quoted. See <https://wiki.hyprland.org> for more examples.
+      '';
+      example = lib.literalExpression ''
+        {
+          general = {
+            after_sleep_cmd = "hyprctl dispatch dpms on";
+            ignore_dbus_inhibit = false;
+            lock_cmd = "hyprlock";
+          };
+
+          listener = [
+            {
+              timeout = 900;
+              on-timeout = "hyprlock";
+            }
+            {
+              timeout = 1200;
+              on-timeout = "hyprctl dispatch dpms off";
+              on-resume = "hyprctl dispatch dpms on";
+            }
+          ];
+        }
+      '';
+    };
+  };
+
+  config = mkIf cfg.enable {
+    xdg.configFile."hypr/hypridle.conf".text =
+      lib.optionalString (cfg.settings != { })
+      (config.lib.hyprlang.toHyprconf cfg.settings 0 true);
```

Do you still want to create the file if the settings are empty? If not, then perhaps you want
```suggestion
    xdg.configFile."hypr/hypridle.conf" = mkIf (cfg.settings != { }) {
      text = config.lib.hyprlang.toHyprconf cfg.settings 0 true;
    }
```